### PR TITLE
Fix 0.7 depwarns

### PR DESCRIPTION
**Pull request by ararslan**

### CODE REVIEW

File: src/QuadGK.jl [hunk 1]
```diff
@@ -16,6 +16,10 @@ using Compat
 using DataStructures, Compat.LinearAlgebra
 import Base: isless, Order.Reverse, AnyDict
 
+if !isdefined(Compat.LinearAlgebra, :rmul!)
+    const rmul! = scale!
+end
```
Comment (stevengj) [inline src/QuadGK.jl]: Would be better to just wait for this to get tagged in Compat...

Comment (ararslan) [reply src/QuadGK.jl]: As far as I know there's been no work on that front to make backwards-compatible versions of `mul!` et al. in Compat, so I figured this was fine for now.

Comment (stevengj) [reply src/QuadGK.jl]: Leaving the deprecation warning for `scale!` in is a reminder that someone should work on this.

Comment (ararslan) [reply src/QuadGK.jl]: Reverted that commit

**Pull request merged**